### PR TITLE
Upgrade boringssl to current chromium-stable

### PR DESCRIPTION
**Pull request by jtattermusch**

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by jtattermusch** [comment]
@muxi   based on some offline information, this breaks the cocoapods podspec, but it's not apparent from the test results. What is the problem exactly and can we do something to catch the problem with automatized tests the next time (relying on knowing we need to check this is problematic)?

CC @matt-kwong

**Comment by muxi** [comment]
Yes I was thinking about the same. To make this automatic we need to reflect the dependency commit change and also boost podspec version in `build.yaml` , so that the podspecs are updated with `generate_project`. I have not had a good way of doing this other than manually changing `build.yaml` every time. If that is acceptable then I can make this happen.

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by muxi** [comment]
Podspec linter passed. cbcdf69 updates it. I'll submit to Cocoapods when this PR merges.

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

File: tools/run_tests/sanity/check_submodules.sh [hunk 1]
```diff
@@ -27,7 +27,7 @@ want_submodules=$(mktemp /tmp/submXXXXXX)
 git submodule | awk '{ print $1 }' | sort > "$submodules"
 cat << EOF | awk '{ print $1 }' | sort > "$want_submodules"
  5b7683f49e1e9223cf9927b24f6fd3d6bd82e3f8 third_party/benchmark (v1.2.0)
- a20bb7ff8bb5057065a2e7941249773f9676cf45 third_party/boringssl (heads/chromium-stable)
+ 70ef9596bbcc11353b9bb8d4e91478694dd21439 third_party/boringssl (fips-20170615-704-g70ef9596)
```
Comment (matt-kwong) [inline tools/run_tests/sanity/check_submodules.sh]: Is `fips-20170615-704-g70ef9596` the name of the branch we're using?

Comment (muxi) [reply tools/run_tests/sanity/check_submodules.sh]: May be the coincide?

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by jtattermusch** [comment]
Known failures:
https://github.com/grpc/grpc/pull/15544

File: tools/run_tests/sanity/check_submodules.sh (hunk 1 above)
Comment (jtattermusch) [reply tools/run_tests/sanity/check_submodules.sh]: yes, they coincide I think. I definitely updated to chromium-stable.  There are rules by which git determines which one is the "primary" name of a given revision, but those priorities aren't set by most projects I've seen AFAIK.

Comment (jtattermusch) [reply tools/run_tests/sanity/check_submodules.sh]: currently chromium-stable is at  70ef9596
https://github.com/google/boringssl/commits/chromium-stable

**Comment by topillar** [comment]
Please backport this patch to 1.12.x & 1.11.x.

Also some thought to BoringSSL,  it's really boring just as the name hint.

**Comment by jtattermusch** [comment]
@topillar this is just routine upgrade of the boringssl submodule - it is not solving any known issue AFAIK,
so there are no plans to backport this change to 1.12.x & 1.11.x.

**Comment by topillar** [comment]
Actually, there are issues related to BoringSSL , pls see #15083.

**Pull request merged**